ops/alert_on_session_errors/program.py

```diff
@@ -0,0 +1,65 @@
+"""Send Slack alerts when AutoKitteh sessions end due to errors.
+
+See the configuration and deployment instructions in the README.md file.
+"""
+
+from datetime import datetime, timedelta, UTC
+import json
+import os
+from urllib.parse import urljoin
+
+from autokitteh.slack import slack_client
+import requests
+from requests import exceptions
+
+
+API_BASE_URL = os.getenv("AUTOKITTEH_API_BASE_URL", "https://api.autokitteh.cloud")
+UI_BASE_URL = os.getenv("AUTOKITTEH_UI_BASE_URL", "https://app.autokitteh.cloud")
+JWT = os.getenv("AUTOKITTEH_AUTH_TOKEN", "")
+SLACK_CHANNEL = os.getenv("SLACK_CHANNEL", "")
+
+slack = slack_client("slack_conn")
+
+
+def on_monitor_schedule(event):
+    """Triggered at the beginning of every minute, so it covers the previous one."""
+    end_time = datetime.now(UTC).replace(second=0, microsecond=0)
+    start_time = end_time - timedelta(minutes=1)
+
+    count = 0
+    for session in reversed(_list_sessions_with_errors()):
+        session_updated = datetime.fromisoformat(session["updatedAt"])
+        if start_time <= session_updated < end_time:
+            count += 1
+            _log_error(session)
+
+    print(f"Found {count} sessions with new errors")
+
+
+def _list_sessions_with_errors():
+    url = urljoin(API_BASE_URL, "autokitteh.sessions.v1.SessionsService/List")
+    headers = {"Content-Type": "application/json"}
+    if JWT:  # Servers in dev mode don't require auth.
+        headers["Authorization"] = "Bearer " + JWT
+
+    resp = requests.post(url, headers=headers, json={"stateType": 3}, timeout=10)
+    print(f"API call's Round Trip Time: {resp.elapsed}")
+    if not resp.ok:
+        resp.raise_for_status()
```

Comment:
The check is redundant, unless you had something specific in mind
https://realpython.com/python-requests/#status-codes
(scroll down a little to the code block example)

```suggestion
    resp.raise_for_status()
```